### DISCUSSION TITLE
Confusing about the momentum residual in ADJunctionParallelChannels1PhaseUserObject

### DISCUSSION THREAD

**Discussion by SomnusYu** [idaholab/moose#25791] [Q&A Modules: Thermal Hydraulics]
Hi @joshuahansel and @licharlot,

I noticed that the computeFluxesAndResiduals() function in ADJunctionParallelChannels1PhaseUserObject calculates momentum residual again, which has been already calculated in ADVolumeJunction1PhaseUserObject. Why do we need to calculate this term twice?

In ADVolumeJunction1PhaseUserObject:
![image](https://github.com/idaholab/moose/assets/118275988/9d396229-dec2-4112-9dcf-31b37bf09229)

In ADJunctionParallelChannels1PhaseUserObject:
![image](https://github.com/idaholab/moose/assets/118275988/3ba3fe29-15d0-46c6-b850-c2524892a3ce)

Thank you very much!

**Comment by GiudGiud** [comment]
Hello

The parallel junction overrides that definition for the first one. So the first one (the base class one) is not called.

Guillaume

**Reply by SomnusYu** [reply]
Hi,

Thanks for your reply. I think the parallel junction did indeed call the base class one.
![image](https://github.com/idaholab/moose/assets/118275988/596cd316-e2ea-4b23-b21d-c44c92b89705)

I think I understand this problem. The symbol of this term in these two classes is contrary. The parallel junction firstly offsets this term in the base class and then recalculates this term as the wall formulation, which is implemented in ADJunctionParallelChannels1PhaseUserObject::finalize() function.

**Reply by GiudGiud** [reply]
Ah good eye.

It doesn't seem very efficient if it's that but it probably does not matter.

**Reply by licharlot** [reply]
Yes, that's it. It was done to avoid duplicating the other terms. I agree it is probably not the best design, but it works.
